core/src/avm2/value.rs

```diff
@@ -570,6 +570,13 @@ impl<'gc> Value<'gc> {
         PrimitiveObject::from_primitive(*self, activation)
     }
 
+    pub fn as_object(&self) -> Option<Object<'gc>> {
```

Comment:
Could we replace `coerce_to_object` with this at some point?

Reply:
Not 100% of cases - I imagine sometimes we might want an actual coercion? (though even less so if we manage to remove `PrimitiveObject` wrapper).
But yes, I imagine that 99% of current `coerce_to_object()` uses actually _expect_ an object, so `as_object()` would apply better there (maybe also an extra function for handling supporting/not supporting/special casing `null/undefined` would be needed too).